bin/registerServerTopology.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+require_once('registerServerTopology-func.php');
+
+/************************ */
+/*     DATA RESOLVING
+/************************ */
+
+/**
+ * Get script params
+ */
+$opt = getopt('u:t:h:n:', ["help::", "dns:", "insecure::"]);
+const SERVER_TYPES = ["poller", "remote", "map", "mbi"];
+
+/**
+ * Format the --help message
+ */
+$helpMessage = <<<'EOD'
+
+Global Options:
+  -u <mandatory>              username of your centreon-web account
+  -t <mandatory>              the server type you want to register:
+            - Poller
+            - Remote
+            - MAP
+            - MBI
+  -h <mandatory>              URL of the Central / Remote Server target platform
+  -n <mandatory>              name of your registered server
+  --help <optional>           get informations about the parameters available
+  --proxy <optional>          provide the differents asked informations
+            - host <mandatory>
+            - port <mandatory>
+            - username <optional>
+            - password <optional>
+  --dns <optional>            provide your DNS instead of automaticaly get server IP
+  --insecure <optional>       allow self-signed certificate
+
+
+EOD;
+
+/**
+ * Display --help message
+ */
+if (isset($opt['help'])) {
+    exit($helpMessage);
+}
+
+/**
+ * Assign options to variables
+ */
+try {
+    if (!isset($opt['u'], $opt['t'], $opt['h'], $opt['n'])) {
+        throw new \InvalidArgumentException(
+            PHP_EOL . 'missing parameter: -u -t -h -n are mandatories:' . PHP_EOL . $helpMessage
+        );
+    }
+
+    $username = $opt['u'];
+    $serverType = in_array(strtolower($opt['t']), SERVER_TYPES)
+        ? strtolower($opt['t'])
+        : false;
+
+    if (!$serverType) {
+        throw new \InvalidArgumentException(
+            "-t must be one of those value"
+            . PHP_EOL . "Poller, Remote, MAP, MBI" . PHP_EOL
```

Comment:
```suggestion
            . PHP_EOL . "TYPE_POLLER, TYPE_REMOTE, TYPE_MAP, TYPE_MBI" . PHP_EOL
```

Reply:
as we store the constant as lowercase, it is most User Friendly to keep this display. 
The user can misstype Upper/lowercase while e.g typing pOLLER, it will be resolve correctly because of the strtolower check l.83.  

If value is incorrect the display message is nicer with Uppercases.